### PR TITLE
[KernelGen][Nvidia] Add hsplit operator with view implementation

### PR DESCRIPTION
**Pull request by ShawnsYing**
## Summary

Add the `hsplit` operator using zero-copy view semantics by delegating to `torch.split`.

## Implementation Details

`hsplit` is a view operator that splits a tensor horizontally (along the width dimension). This implementation:

- Uses zero-copy view semantics via `torch.split`
- Handles both integer (equal splits) and list (custom splits) arguments
- Supports all tensor dimensions (1D splits at dim=0, 2D+ splits at dim=1)
- No custom kernel needed - delegates to PyTorch's optimized `split`

## Verification

**Four-gate testing (CLAUDE.md §10A.2.1)**:
1. ✅ Pre-commit: PASSED
2. ✅ Direct dispatch: Confirmed "GEMS HSPLIT" log
3. ✅ GPU tests: 30/30 PASSED (11.22s)
4. ✅ Quick-CPU tests: 6/6 PASSED (1.61s)
5. ✅ Benchmark: PASSED (27.41s)

## Test Coverage

- Integer splits: equal-sized chunks
- List splits: custom split sizes
- Various tensor dimensions (1D, 2D, 3D)
- Edge cases (single split, unequal sizes)

## Files Changed

- `src/flag_gems/ops/hsplit.py`: view implementation (new, 37 lines)
- `conf/operators.yaml`: operator registration (15 lines)
- `src/flag_gems/ops/__init__.py`: export (2 lines)
- `src/flag_gems/__init__.py`: dispatch registration (1 line)
- `tests/test_hsplit.py`: comprehensive tests (new, 69 lines)
- `benchmark/test_hsplit.py`: benchmarks (new, 53 lines)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/flag_gems/ops/hsplit.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Generated by KernelGen: https://github.com/flagos-ai/KernelGen
+import logging
+from typing import List, Union
+
+import torch
+
+logger = logging.getLogger(__name__)
+
+
+def hsplit(
```
Comment (Caeruleann) [inline src/flag_gems/ops/hsplit.py:24]: So, it's not a triton operator?

**Review by Caeruleann — APPROVED** [review]
